api/tests/unit/domain/usecases/create-or-update-stage-collection_test.js

```
@@ -0,0 +1,205 @@
+          level: undefined,
+          title: 'first stage',
+          message: 'initial first message',
+          prescriberTitle: 'initial first prescriber title',
+          prescriberDescription: 'initial first description',
+        }),
+        domainBuilder.buildStage({
+          id: 2,
+          threshold: 10,
+          level: undefined,
+          title: 'second stage',
+          message: 'initial second message',
+          prescriberTitle: 'initial second prescriber title',
+          prescriberDescription: 'initial second description',
+        }),
+      ];
+
+      stageCollectionForTargetProfileRepository = {
+        getByTargetProfileId: sinon.stub().resolves(
+          domainBuilder.buildStageCollectionForTargetProfileManagement({
+            stages,
+          }),
+        ),
+        update: sinon.stub(),
+      };
+
+      targetProfileForAdminRepository = {
+        get: sinon.stub().resolves({
+          id: 1,
+          hasLinkedCampaign: true,
+          stageCollection: { stages },
+        }),
+      };
+
+      targetProfile = domainBuilder.buildTargetProfileForAdmin();
+    });
+
+    it('should be possible to update all fields that are not threshold or level', async function () {
+      // given
+      const stagesFromPayload = [
+        {
+          ...stages[0],
+          title: 'updated title',
+          message: 'updated message',
+          prescriberTitle: 'updated prescriber title',
+          prescriberDescription: 'updated prescriber description',
+        },
+        { ...stages[1] },
+      ];
+
+      // when
+      await createOrUpdateStageCollection({
+        stagesFromPayload,
+        targetProfileId: targetProfile.id,
+        stageCollectionForTargetProfileRepository,
+        targetProfileForAdminRepository,
+      });
+
+      // then
+      expect(stageCollectionForTargetProfileRepository.update).to.have.been.called;
+    });
+
+    it('should not be possible to create a stage', async function () {
```

`should throw an error if there is a new stage`